validate/vmspec.go

```diff
@@ -0,0 +1,274 @@
+// Copyright 2017 The go-interpreter Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+package validate
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"github.com/go-interpreter/wagon/wasm"
+	"github.com/go-interpreter/wagon/wasm/leb128"
+	ops "github.com/go-interpreter/wagon/wasm/operators"
+	"io"
+)
+
+type frameType uint8
+
+const (
+	frameTypeFunction frameType = 0x5f
+	frameTypeBlock    frameType = 0x5e
+	frameTypeIf       frameType = 0x5d
+	frameTypeElse     frameType = 0x5c
+	frameTypeLoop     frameType = 0x5b
+)
+
+type mockSpecVM struct {
+	opdStack   []wasm.ValueType
+	ctrlStack  []ctrlFrame
+	origLength uint32
+	code       *bytes.Reader
+	curFunc    *wasm.FunctionSig
+}
+
+type ctrlFrame struct {
+	labelTypes  wasm.ValueType
```

Comment:
single value is enough here? spec use `[]wasm.ValueType`

Reply:
currenly only consider MVP